src/plugins/saved_objects/README.md

```diff
@@ -0,0 +1,119 @@
+# Saved Object
+
+The saved object plugin provides all the core services and functionalities of the saved objects, and is utilized by many other plugins such as visualizaton plugin, dashboard plugin and wizard plugin etc.
+
+## Save Relationships to Index Pattern
+
+The relationships to index patterns are saved using the kibanaSavedObjectMeta attribute and the references array structure. Functions from the data plugin are utilized by the saved object plugin to manage this index pattern relationships. 
+
+A standard saved object and its index pattern relationship:
+
+```.ts
+
+"kibanaSavedObjectMeta" : {
+    "searchSourceJSON" : """{"filter":[],"query":{"query":"","language":"kuery"},"indexRefName":"kibanaSavedObjectMeta.searchSourceJSON.index"}"""
+      }
+    },
+    "type" : "visualization",
+    "references" : [
+      {
+          "name" : "kibanaSavedObjectMeta.searchSourceJSON.index",
+          "type" : "index-pattern",
+          "id" : "90943e30-9a47-11e8-b64d-95841ca0b247"
+      }
+    ],
+
+```
+
+### Saving a saved object
+
+When saving a saved object and its relationship to the index pattern: 
+
+1. A saved object will be built using `buildSavedObject` function, services such as hydrating index pattern, initializing and serializing the saved object are set and configs such as saved object id, migration version are defined.
```

Comment:
```suggestion
1. A saved object will be built using `buildSavedObject` function, which hydrates index pattern, initializes  the saved object, and defines configs such as saved object id and migration version.
```

Where is `buildSavedObject` defined? If possible, it can be nice to link to a permalink in the source code.

Reply:
Yes! i will add permalink throughout the doc